### PR TITLE
[html-elements] Move BR to Rules

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

`test-suite` was failing with an ominous error: **Views nested within a <Text> must have a width and height**.

# How

This took a while to debug (it seemed like the error was thrown while testing `Nav`, but in fact the logging was just slower than testing and it was thrown when testing `BR`). `test-suite` expects `BR` to be a view element, so I made it.

https://github.com/expo/expo/blob/90a1e11151ce0689ea0bf6d89b55cf37ceeff47b/apps/test-suite/tests/HTML.js#L74-L80

# Test Plan

Test no longer fails with **Views nested within…**.

### CODE REVIEW

File: packages/html-elements/src/elements/Text.tsx [hunk 1]
```diff
@@ -34,9 +34,9 @@ export const BlockQuote = forwardRef(({ style, cite, ...props }: BlockQuoteProps
   return <View {...props} style={[styles.blockQuote, style]} ref={ref} />;
 }) as ComponentType<BlockQuoteProps>;
 
-export const BR = forwardRef(({ style, ...props }: TextProps, ref) => {
-  return <Text {...props} style={[styles.br, style]} ref={ref} />;
-}) as ComponentType<TextProps>;
+export const BR = forwardRef(({ style, ...props }: ViewProps, ref) => {
```
Comment (EvanBacon) [inline packages/html-elements/src/elements/Text.tsx]: Should probably move this to the `Rules.tsx` files (same with the tests).

**Review by EvanBacon — CHANGES_REQUESTED** [review]
Tests will need to be updated (`yarn test -u`)

**Comment by sjchmiela** [comment]
- moved to `Rules`,
- reverted backing view to `Text`
- added a web-compatible implementation in `Rules.web.tsx` that uses native `<br />`
- added a special test case in `test-suite`, since neither `<Element>Test contents</Element>` nor `<Element><View /></Element>` applies to `<BR />` element.
- added a BR-ed paragraph to NCL to be able to test its functionality visually

**Comment by sjchmiela** [comment]
Fixed by @esamelson in https://github.com/expo/expo/pull/8642.

**Pull request closed without merging**